mule-extensions-api/src/main/java/org/mule/runtime/extension/internal/loader/validator/NameClashModelValidator.java

```diff
@@ -596,6 +598,20 @@ public String toString() {
     }
   }
 
+  private boolean areTypesEqual(MetadataType type, MetadataType otherType) {
+    String typeId = getTypeId(type);
+    String otherTypeId = getTypeId(otherType);
+
+    if (typeId != null && otherTypeId != null) {
```

Comment:
this should first check for the `ClassInformationTypeAnnotation` and if it exists check for the types to be assignable

Reply:
To clarify,
 I should check that if both have the annotation. If both have it, I check if the annotations are equals to each other?

After that, the strategy remain the same.

Reply:
no. If they both have it, you have to consider that the two types may be equal or that one may be assignable from the other (`Apple extends Fruit`)

Reply:
Should I get the class loader to load the class given the class name and check if A is assingable to B or B is assignable to A then?

Reply:
If the ExtensionModel has a ClassLoaderModelProperty available then yes, that'd be ideal. Otherwise, you can only compare the classnames.